code/game/verbs/suicide.dm

```diff
@@ -1,84 +1,52 @@
-/mob/var/suiciding = 0
+/mob/living/var/suiciding = 0
 
-/mob/living/carbon/human/verb/suicide() /// At best, useful for admins to see if it's being called.
-	set hidden = 1
-
-	if (stat == DEAD)
-		to_chat(src, "You're already dead!")
-		return
-
-	if (!ticker)
-		to_chat(src, "You can't commit suicide before the game starts!")
-		return
-	
-	to_chat(src, "<span class='warning'>No. Adminhelp if there is a legitimate reason, and please review our server rules.</span>")
-	message_admins("[ckey] has tried to trigger the suicide verb as human, but it is currently disabled.")
-
-/mob/living/carbon/brain/verb/suicide()
+/mob/living/verb/suicide()
 	set hidden = 1
 
 	if (stat == 2)
 		to_chat(src, "You're already dead!")
-		return
-
+		return FALSE
 	if (!ticker)
 		to_chat(src, "You can't commit suicide before the game starts!")
-		return
+		return FALSE
 
 	if (suiciding)
 		to_chat(src, "You're already committing suicide! Be patient!")
-		return
+		return FALSE
+
+	return (alert("Are you sure you want to commit suicide?", "Confirm Suicide", "Yes", "No") == "Yes")
+
 
-	var/confirm = alert("Are you sure you want to commit suicide?", "Confirm Suicide", "Yes", "No")
+/mob/living/carbon/human/suicide() /// At best, useful for admins to see if it's being called.
+	..()
+	to_chat(src, "<span class='warning'>No. Adminhelp if there is a legitimate reason, and please review our server rules.</span>")
+	message_admins("[ckey] has tried to trigger the suicide verb as human, but it is currently disabled.")
 
-	if(confirm == "Yes")
+/mob/living/carbon/brain/suicide()

```

Comment:
Can't they just ghost instead of this weird-ass message and proc? If you're going to hide the verb and forbid it generally it seems odd for it to have a legitimate use.